ApplicationLibCode/Commands/WellPathCommands/RicWellPathsUnitSystemSettingsUi.cpp

```diff
@@ -27,23 +27,8 @@ RicWellPathsUnitSystemSettingsUi::RicWellPathsUnitSystemSettingsUi()
 {
     CAF_PDM_InitObject( "RimWellPathsUnitSystemSettings" );
 
-    CAF_PDM_InitFieldNoDefault( &unitSystem, "UnitSystem", "Unit System" );
-}
-
-//--------------------------------------------------------------------------------------------------
-///
-//--------------------------------------------------------------------------------------------------
-QList<caf::PdmOptionItemInfo> RicWellPathsUnitSystemSettingsUi::calculateValueOptions( const caf::PdmFieldHandle* fieldNeedingOptions )
-{
-    QList<caf::PdmOptionItemInfo> options;
-    if ( fieldNeedingOptions == &unitSystem )
-    {
-        options.push_back(
-            caf::PdmOptionItemInfo( caf::AppEnum<RiaDefines::EclipseUnitSystem>::uiText( RiaDefines::EclipseUnitSystem::UNITS_METRIC ),
-                                    RiaDefines::EclipseUnitSystem::UNITS_METRIC ) );
-        options.push_back(
-            caf::PdmOptionItemInfo( caf::AppEnum<RiaDefines::EclipseUnitSystem>::uiText( RiaDefines::EclipseUnitSystem::UNITS_FIELD ),
-                                    RiaDefines::EclipseUnitSystem::UNITS_FIELD ) );
-    }
-    return options;
+    CAF_PDM_InitField( &unitSystem, "UnitSystem", RiaDefines::EclipseUnitSystem::UNITS_METRIC, "Unit System" );
+    caf::AppEnum<RiaDefines::EclipseUnitSystem>::setEnumSubset( unitSystem.keyword(),
```

Comment:
Would sending in the field itself, not the keyword, make more sense to the developers using it? -> &unitSystem, not unitSystem.keyword()

Reply:
Yes, agree. Will fix.